src/H5PLpath.c

```diff
@@ -546,7 +548,220 @@ H5PL__get_path(unsigned int idx)
     return H5PL_paths_g[idx];
 done:
     FUNC_LEAVE_NOAPI(ret_value)
-} /* end H5PL__replace_path() */
+} /* end H5PL__get_path() */
+
+/*-------------------------------------------------------------------------
+ * Function:    H5PL__path_table_iterate
+ *
+ * Purpose:     Iterates over all the plugins in the plugin path table and
+ *              calls the specified callback function on each plugin found.
+ *
+ * Return:      H5_ITER_CONT if all plugins are processed successfully
+ *              H5_ITER_STOP if short-circuit success occurs while
+ *                  processing plugins
+ *              H5_ITER_ERROR if an error occurs while processing plugins
+ *
+ *-------------------------------------------------------------------------
+ */
+herr_t
+H5PL__path_table_iterate(H5PL_iterate_type_t iter_type, H5PL_iterate_t iter_op, void *op_data)
+{
+    unsigned int u;
+    herr_t       ret_value = H5_ITER_CONT;
+
+    FUNC_ENTER_PACKAGE
+
+    for (u = 0; (u < H5PL_num_paths_g) && (ret_value == H5_ITER_CONT); u++)
+        ret_value = H5PL__path_table_iterate_process_path(H5PL_paths_g[u], iter_type, iter_op, op_data);
+
+    if (ret_value < 0)
+        HGOTO_ERROR(H5E_PLUGIN, H5E_BADITER, H5_ITER_ERROR, "can't iterate over plugins in plugin path '%s'",
+                    H5PL_paths_g[u]);
+
+done:
+    FUNC_LEAVE_NOAPI(ret_value);
+} /* end H5PL__path_table_iterate() */
+
+/*-------------------------------------------------------------------------
+ * Function:    H5PL__path_table_iterate_process_path
+ *
+ * Purpose:     Iterates over all the plugins within a single plugin path
+ *              entry in the plugin path table and calls the specified
+ *              callback function on each plugin found. Two function
+ *              definitions are for Unix and Windows.
+ *
+ * Return:      H5_ITER_CONT if all plugins are processed successfully
+ *              H5_ITER_STOP if short-circuit success occurs while
+ *                  processing plugins
+ *              H5_ITER_ERROR if an error occurs while processing plugins
+ *
+ *-------------------------------------------------------------------------
+ */
+#ifndef H5_HAVE_WIN32_API
+static herr_t
+H5PL__path_table_iterate_process_path(const char *plugin_path, H5PL_iterate_type_t iter_type,
```

Comment:
I'm not a big fan of needing two versions of this routine, but it would take too long for me to refactor the system-dependent code like we discussed last time this was reviewed and I don't want to risk breaking the existing plugin code by getting that wrong. In the interest of time, I'm leaving sleeping dogs lie here.

Reply:
I can revisit this when I work on the VFD plugins early next year.